ADR/XXXX-pipeline-caching-feature-flag.md

```diff
@@ -0,0 +1,79 @@
+# XXXX. Pipeline Caching Feature Flag Configuration
+
+Date: 2025-10-23
+
+## Status
+
+Proposed
+
+## Context
+
+Following the implementation of container build layer caching via Squid HTTP proxies as described in [ADR 0047](0047-caching-for-container-build-layers.md), we need to provide users with flexible control over when and how the proxy caching is enabled for their pipelines. Users need the ability to:
+
+* Enable or disable HTTP proxy use for individual pipelines without knowing technical details about the proxy configuration
+* Control proxy usage at the namespace level for all pipelines running in that namespace
+* Control proxy usage at the cluster level for all pipelines in the cluster
+* Apply GitOps processes for this configuration, similar to other Konflux configuration elements
+
+The current design requires users to manually configure `HTTP_PROXY` and `NO_PROXY` parameters to the buildah task, which exposes implementation details and creates a poor user experience. We need a more user-friendly configuration mechanism that abstracts away the technical details while providing appropriate levels of control.
+
+## Decision
+
+We will implement a hierarchical feature flag system for controlling pipeline caching proxy usage with the following configuration levels and precedence:
+
+### Configuration Levels
+
+1. **Pipeline Level**: An `enable-cache-proxy` parameter that can be set in individual pipeline definitions
+2. **Namespace Level**: An `enable-cache-proxy` value in the `konflux-config` config map in the pipeline's namespace
+3. **Cluster Level**: An `enable-cache-proxy` value in the `cluster-config` config map in the `konflux-info` namespace
+
+### Precedence Order
+
+Configuration takes effect based on the following precedence (highest to lowest):
+
+1. Cluster-level configuration (`cluster-config` in `konflux-info` namespace)
+2. Namespace-level configuration (`konflux-config` in pipeline's namespace)
+3. Pipeline-level configuration (`ENABLE_CACHE_PROXY` parameter)
```

Comment:
I think it will be very confusing to users if they have a pipeline parameter that says one thing, but actually does completely different stuff because different settings are used. We should avoid such behavior as much as possible.

Reply:
I'm already confused by this.

I may understand need of global turn-off of proxy, but  local namespace overriding pipeline config is too unexpected.

IMHO, we should have a separate admin flag in cluster config that is true/false/defer, and keep namespace with lower precedence than pipeline (and allow only true/false)

Reply:
I'll think this is satisifed now.

* The user elects to use the cache with a param on their pipeline.
* The admin can disallow use of the cache globally.

The user sets *enable* and *disable* (like requests) and the admin sets *allow* and *disallow* (like a limit or quota).